pkg/agent/status_manager.go

```diff
@@ -131,6 +133,22 @@ func (mgr *Manager) updateClusterStatus(ctx context.Context, namespace, clusterI
 			return false, nil
 		}
 
+		newLabels := mgr.getNewManagedClusterLabels()
+		if !reflect.DeepEqual(mgr.managedCluster.GetLabels(), newLabels) {
+			mgr.managedCluster.Labels = newLabels
+			mcls, lastError = client.ClustersV1beta1().ManagedClusters(namespace).Update(ctx, mgr.managedCluster, metav1.UpdateOptions{})
```

Comment:
We can use method `PATCH` to update labels.

Reply:
please take another look